app/models/user.rb

```diff
@@ -20,4 +20,14 @@ def send_registered_email
     RegistrationMailer.registration_email(self).deliver_now
   end
 
+
+  validate :validate_donor_fields
+
+  def validate_donor_fields
+    if role_type == 'Donor'
```

Comment:
That method doesn't seem to have loaded (or something like that) by then - `if donor?` always returns false, even when the user is a donor, which makes the validation useless